projects/experimental/components/badge/badge.component.ts

```diff
@@ -0,0 +1,58 @@
+import {AnimationOptions} from '@angular/animations';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    HostBinding,
+    Inject,
+    Input,
+} from '@angular/core';
+import {TUI_PLATFORM, TuiPlatform} from '@taiga-ui/cdk';
+import {
+    MODE_PROVIDER,
+    TUI_ANIMATION_OPTIONS,
+    TUI_MODE,
+    TuiBrightness,
+    tuiPop,
+    TuiSizeS,
+    TuiSizeXL,
+} from '@taiga-ui/core';
+import {TuiBadgeAppearance} from '@taiga-ui/experimental/types';
+import {Observable} from 'rxjs';
+
+@Component({
+    selector: 'tui-badge',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    templateUrl: './badge.template.html',
+    styleUrls: ['./badge.style.less'],
+    providers: [MODE_PROVIDER],
+    animations: [tuiPop],
+    host: {
+        '[tabIndex]': 'hoverable ? 0 : -1',
+        '($.data-mode.attr)': 'mode$',
+    },
+})
+export class TuiBadgeComponent {
+    @Input()
+    @HostBinding('attr.data-size')
+    size: TuiSizeS | TuiSizeXL = 'l';
+
+    @Input()
+    @HostBinding('attr.data-appearance')
+    appearance: TuiBadgeAppearance = 'default';
+
+    @Input()
+    @HostBinding('class._hoverable')
+    hoverable = false;
+
+    @HostBinding('attr.data-platform')
+    platform = this.tuiPlatform;
```

Comment:
Unnecessary prop, you can move it to `host`. We can actually move all bindings to host and drop `@HostBinding`, I believe it makes bundle size a little bit smaller and uniformity within component is a good thing.